### PR TITLE
MM-36881 Bring back refresh control for channels and threads

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Brings back the pull to refresh functionality on channels and threads, and actually fetch the latest posts in the channel or the posts that belong to the thread depending on the screen the user is in.

#### Ticket Link
JIRA: https://mattermost.atlassian.net/browse/MM-36881

#### Release Note
```release-note
Added back pull to refresh
```

#### Test Note
on Android we should test, check and double check that there is no performance degradation while scrolling the post list.

### CODE REVIEW

File: patches/react-native+0.64.2.patch [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
++        let inverted;
++        if (inner.scaleY) {
++          inverted = {scaleY: -1};
++          delete inner['scaleY']
++        }
++
+         return React.cloneElement(
+           refreshControl,
+-          {style: [baseStyle, outer]},
++          {style: [baseStyle, outer, inverted]},
```
Comment (enahum) [inline patches/react-native+0.64.2.patch:18]: This is the patch that allows us to bring back the pull to refresh indicator on android that respects the inverted control

**Review by anurag6713 — APPROVED** [review]
LGTM!

**Comment by esethna** [comment]
@svelle would you be able to test this PR on your Android device with an eye out for performance degradations? The build should appear in the QA channel once it's up

**Review by esethna — APPROVED** [review]
Tested on iOS, works great, thanks @enahum!!

**Review by josephbaylon — APPROVED** [review]
LGTM. Refresh now appears in the channels when user pulls up the screen. Also tested scrolling of post list on both iOS and Android and didn't experience performance issues.

I'll let @svelle do the final verification on his Android device.

**Pull request merged**